src/http_server.cc

```diff
@@ -108,6 +108,46 @@ AddContentTypeHeader(evhtp_request_t* req, const char* type)
       req->headers_out, evhtp_header_new(kContentTypeHeader, type, 1, 1));
 }
 
+TRITONSERVER_Error*
+SetTritonParameterFromJsonParameter(
+    const std::string& parameter,
+    triton::common::TritonJson::Value& params_json,
+    TRITONSERVER_InferenceRequest* irequest)
+{
+  triton::common::TritonJson::Value value;
+  if (!params_json.Find(parameter.c_str(), &value)) {
+    return TRITONSERVER_ErrorNew(
+        TRITONSERVER_ERROR_INTERNAL,
+        ("parameter key '" + parameter + "' was not found in the JSON")
+            .c_str());
+  }
+
+  if (value.IsString()) {
+    std::string string_value;
+    RETURN_IF_ERR(value.AsString(&string_value));
```

Comment:
One option would be to use RETURN_MSG_IF_ERR on the json conversions and form a message with the name of the variable.

We could create a single string with the name and use it.